File: backend/src/models/campaign.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize)]
pub struct Campaign {
    pub id: Uuid,
    pub title: String,
    pub description: String,
    pub creator_id: Uuid,
    pub target_amount: f64,
    pub current_amount: f64,
    pub location_lat: f64,
    pub location_lng: f64,
    pub created_at: DateTime<Utc>,
    pub ends_at: DateTime<Utc>,
}

#[derive(Debug, Serialize)]
pub struct CampaignMetadata {
    pub id: Uuid,
    pub title: String,
    pub description: String,
    pub target_amount: f64,
    pub current_amount: f64,
    pub location_lat: f64,
    pub location_lng: f64,
    pub ends_at: DateTime<Utc>,
}

File: frontend/lib/models/campaign.dart
class Campaign {
  final String id;
  final String title;
  final String description;
  final String organizer;
  final double goalAmount;
  final double currentAmount;
  final DateTime? createdAt;
  final String imageUrl;

  const Campaign({
    required this.id,
    required this.title,
    required this.description,
    required this.organizer,
    required this.goalAmount,
    required this.currentAmount,
    this.createdAt,
    required this.imageUrl,
  });
}
